File: java/src/main/java/problems201_300/RemoveLinkedListElements.java
package problems201_300;

import concepts.SinglyListNode;

enum RemoveLinkedListElements {
    DEFAULT {
        @Override
        SinglyListNode removeElements(SinglyListNode head, int val) {
            SinglyListNode newHead = new SinglyListNode();
            SinglyListNode currentNew = newHead;

            SinglyListNode current = head;
            while (current != null) {
                if (current.val != val) {
                    currentNew.next = new SinglyListNode(current.val);
                    currentNew = currentNew.next;
                }
                current = current.next;
            }
            return newHead.next;
        }
    };

    abstract SinglyListNode removeElements(SinglyListNode head, int val);
}

File: kotlin/src/main/kotlin/problems201_300/RemoveLinkedListElements.kt
package problems201_300

import concepts.SinglyListNode

enum class RemoveLinkedListElements {
    DEFAULT {
        override fun removeElements(head: SinglyListNode?, `val`: Int): SinglyListNode? {
            val newHead = SinglyListNode()
            var currentNew = newHead

            var current = head
            while (current != null) {
                if (current.`val` != `val`) {
                    currentNew.next = SinglyListNode(current.`val`)
                    currentNew = currentNew.next!!
                }
                current = current.next
            }
            return newHead.next
        }
    },
    ;

    abstract fun removeElements(head: SinglyListNode?, `val`: Int): SinglyListNode?
}
